docs/userGuide/syntax/code.mbdf

```diff
@@ -58,42 +58,63 @@ function add(a, b) {
 
-For ranges of lines, join the two line numbers with a dash sign (`-`).
+Type | Format | Example
+-----|--------|--------
+**Full text highlight**<br>Highlights the entirety of the text portion of the line | The line numbers as-is (subject to the starting line number set in `start-from`). | `3`, `5`
+**Substring highlight**<br>Highlights _all_ occurrences of a substring in the line | `lineNumber[part]`<br><br>_Limitations_: `part` must be wrapped in quotes. If `part` contains a quote, escape it with a backslash (`\`). | `3['Inventory']`,`4['It\'s designed']`
+**Character-bounded highlight**<br>Highlights a specific range of characters in the line | `lineNumber[start:end]`, highlights from character position `start` up to (but not including) `end`.<br><br>Character positions start from `0` as the first non-whitespace character, upwards.<br><br>Omit either `start`/`end` to highlight from the start / up to the end, respectively. | `19[1:5]`,`30[10:]`,`35[:20]`
+**Word-bounded highlight**<br>Highlights a specific range of words in the line | `lineNumber[start::end]`, highlights from word position `start` up to (but not including) `end`.<br><br>Word positions start from `0` as the first word (sequence of non-whitespace characters), upwards<br><br>Omit either `start`/`end` to highlight from the start / up to the end, respectively. | `5[2::4]`,`9[1::]`,`11[::5]`
+**Full line highlight**<br>Highlights the entirety of the line | `lineNumber[:]` | `7[:]`
 
-**Whole-line highlighting**
+Not only a single line, MarkBind is also capable of highlighting ranges of lines in various ways. In general, the syntax
+for range highlighting consists of two single line highlight rules as listed above joined by a dash (`-`).
 
-If you wish to highlight a full line (including whitespaces) or ranges of it, you can leverage MarkBind's own _line-slice_ syntax. Line-slices are in the form of `lineNumber[:]`, e.g. `2[:]`.
-    
-This variant's format is very similar to the previous, but instead use line-slices rather than line numbers.
-
-For ranges, you only need to use line-slices on either ends.
+Type | Format | Example
+-----|--------|--------
+**Ranged full text highlight**<br>Highlights only the text portion of the lines within the range | `lineStart-lineEnd` | `2-4`
+**Ranged full line highlight**<br>Highlights the entirety of the lines within the range | `lineStart[:]-lineEnd` or `lineStart-lineEnd[:]` | `1[:]-5`,`10-12[:]`
+**Ranged character-bounded highlight**<br>Highlights the text portion of the lines within the range, but starts/ends at an arbitrary character | `lineStart[start:]-lineEnd` or `lineStart-lineEnd[:end]` | `3[2:]-7`, `4-9[:17]`
+**Ranged word-bounded highlight**<br>Highlights the text portion of the lines within the range, but starts/ends at an arbitrary word | `lineStart[start::]-lineEnd` or `lineStart-lineEnd[::end]` | `16[1::]-20`,`22-24[::3]`
 
 <include src="codeAndOutputCode.md" boilerplate >
```

Comment:
let's shift this up before `the value of highlight-lines...` so the user has a brief idea what the usage is like first :-)

Reply:
might have missed one ^ @ryoarmanda 
disregard if it looks stranger

Reply:
I did this already @ang-zeyu, I just reworded `the value of highlight-lines...` to the ones in line `85-86`

Reply:
> this

was suggesting the reverse - moving the example up (the `<include>` tag), so the user gets a brief overview of the entire syntax usage first before the details

Reply:
Ahh okay, will rebase the fix shortly

Reply:
Done, the change is reverted in the force push :)